tools/distpartitioning/utils.py

```diff
@@ -47,6 +50,34 @@ def read_json(json_file):
 
     return val
 
+def get_ntype_featnames(ntype_name, schema): 
+    """
+    Retrieves node feature names for a given node_type
+
+    Parameters:
+    -----------
+    ntype_name : string
+        a string specifying a node_type name
+
+    schema : dictionary
+        metadata json object as a dictionary, which is read from the input
+        metadata file from the input dataset
+
+    Returns:
+    --------
+    list : 
+        a list of feature names for a given node_type
+    """
+    ntype_dict = schema["node_data"]
+    if (ntype_name in ntype_dict):
```

Comment:
if node type should always exist, we should use asset to make sure it doesn't fail silently.

Reply:
In this case ntype_name is NOT always guaranteed to exist in the node-feature portion of the metadata. So this code is needed for this particular case. Basically not all node types will have node feature data associated with them.